tianshou/utils/log_tools.py

```diff
@@ -0,0 +1,62 @@
+import threading
+from typing import Dict, Optional
+from torch.utils import tensorboard
+
+
+class SummaryWriter(tensorboard.SummaryWriter):
+    """A more convenient Summary Writer(`tensorboard.SummaryWriter`).
+
+    You can get the same instance of summary writer everywhere after you
+    created one.
+    ::
+
+        >>> writer1 = SummaryWriter.get_instance(
+                key="first", log_dir="log/test_sw/first")
+        >>> writer2 = SummaryWriter.get_instance()
+        >>> writer1 is writer2
+        True
+        >>> writer4 = SummaryWriter.get_instance(
+                key="second", log_dir="log/test_sw/second")
+        >>> writer5 = SummaryWriter.get_instance(key="second")
+        >>> writer1 is not writer4
+        True
+        >>> writer4 is writer5
+        True
+    """
+
+    _mutex_lock = threading.Lock()
```

Comment:
The mutex would be different if used in conjunction with multiprocessing. The only way to circumvent this limitation is to add the lock in the constructor. I don't know if  `traitlet` is handling this properly but I guess so.

Reply:
> I don't know if traitlet is handling this properly but I guess so.

nope

Reply:
I don't think there's any situation where singleton summarywriter need to be used in multi-process. If that happened, then different summarywriters were supposed to be used or only the main process should take responsibility for logging. The mutex here is mainly used to avoid creating multiple instances with same key in multi-thread.